### PR TITLE
delete editor tools from user settings

### PR DESCRIPTION
**Pull request by e11sy**
- added ability to delete not default tools from user settings

### CODE REVIEW

File: src/application/i18n/messages/en.json [hunk 1]
```diff
@@ -32,5 +32,8 @@
   "marketplace": {
     "title": "Marketplace",
     "listOfTools": "Tools"
+  },
+  "editorTools": {
+    "delete": "delete"
```
Comment (neSpecc) [inline src/application/i18n/messages/en.json:37]: Lets move to the `userSettings` and call `Unistall`

File: src/infrastructure/user.repository.ts [hunk 2]
```diff
@@ -76,5 +76,17 @@ export default class UserRepository extends Repository<UserStore, UserStoreData>
...
+   * Delets a tool from the user (marketplace mock)
+   *
+   * @param id - tool id
+   */
+  public async deleteTool(id: string): Promise<void> {
+    const response = await this.transport.delete<{toolId: string}>('/user/editor-tools', {
+      toolId: id,
+    });
+
+    console.log('Delete tool response', response);
```
Comment (neSpecc) [inline src/infrastructure/user.repository.ts:90]: why do we need it?

File: src/presentation/pages/Settings.vue [hunk 3]
```diff
@@ -55,6 +62,15 @@ async function userLogout() {
   });
 }
 
+/**
+ * Deletes tool from the user
+ *
+ * @param toolId - id of the tool
+ */
+async function userDeleteTool(toolId: string) {
+  await deleteTool(toolId);
```
Comment (neSpecc) [inline src/presentation/pages/Settings.vue:71]: what will happen next? Button text should be changed to "Uninstalled" and it should became inactive

File: src/domain/user.repository.interface.ts [hunk 4]
```diff
@@ -29,10 +29,18 @@ export default interface UserRepositoryInterface {
...
+
   /**
    * Adds a tool to the user (marketplace mock)
    *
    * @param id - tool id
    */
   addTool: (id: string) => void;
+
+  /**
+   * Deletes a tool from the user (marketplace mock)
```
Comment (slaveeks) [inline src/domain/user.repository.interface.ts:41]: It is not mock for now

File: src/domain/user.service.ts [hunk 5]
```diff
@@ -52,4 +52,13 @@ export default class UserService {
   public addTool(id: string): void {
     this.repository.addTool(id);
   }
+
+  /**
+   * Deletes a tool from the user (marketplace mock)
```
Comment (slaveeks) [inline src/domain/user.service.ts:57]: Here too

File: src/presentation/pages/Settings.vue (hunk 3 above)
Comment (e11sy) [reply src/presentation/pages/Settings.vue:71]: and after reload of the page it should be removed from user settings?

Comment (neSpecc) [reply src/presentation/pages/Settings.vue:71]: yes

File: src/application/i18n/messages/en.json [hunk 6]
```diff
@@ -13,7 +13,8 @@
   "userSettings": {
     "title": "User Settings",
     "changeTheme": "Change theme",
-    "userEditorTools": "User editor tools"
+    "userEditorTools": "User editor tools",
+    "delete": "delete"
```
Comment (neSpecc) [inline src/application/i18n/messages/en.json:17]: key is unclear. And value as well.

File: src/presentation/pages/Settings.vue [hunk 7]
```diff
@@ -7,6 +7,11 @@
   >
     <li>
       {{ tool.title }}
+      <Button
+        v-if="tool.isDefault === false"
+        :text="t('userSettings.delete')"
```
Comment (neSpecc) [inline src/presentation/pages/Settings.vue:12]: Uninstall

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: src/application/i18n/messages/en.json [hunk 8]
```diff
@@ -13,7 +13,8 @@
   "userSettings": {
     "title": "User Settings",
     "changeTheme": "Change theme",
-    "userEditorTools": "User editor tools"
+    "userEditorTools": "User editor tools",
+    "deleteEditorTool": "Uninstall"
```
Comment (neSpecc) [inline src/application/i18n/messages/en.json:17]:
```suggestion
    "uninstallEditorTool": "Uninstall"
```

File: src/presentation/pages/Settings.vue [hunk 9]
```diff
@@ -54,6 +61,15 @@ async function userLogout() {
...
   });
 }
+
+/**
+ * Deletes tool from the user
+ *
+ * @param toolId - id of the tool
+ */
+async function userRemoveTool(toolId: string) {
+  await removeTool(toolId);
```
Comment (neSpecc) [inline src/presentation/pages/Settings.vue:71]: Will the list be updated after delete?

File: src/application/i18n/messages/en.json (hunk 6 above)
Comment (e11sy) [reply src/application/i18n/messages/en.json]: changed

File: src/presentation/pages/Settings.vue (hunk 9 above)
Comment (e11sy) [reply src/presentation/pages/Settings.vue]: yes

File: src/presentation/pages/Settings.vue [hunk 10]
```diff
@@ -54,6 +61,17 @@ async function userLogout() {
...
   });
 }
+
+/**
+ * Deletes tool from the user
+ *
+ * @param toolId - id of the tool
+ */
+async function uninstallClicked(toolId: string) {
+  if (window.confirm('Do you really want to delete thid tool?')) {
```
Comment (slaveeks) [inline src/presentation/pages/Settings.vue:71]: use i18n

File: src/application/services/useUserSettings.ts [hunk 11]
```diff
@@ -18,7 +18,7 @@ interface UseUserSettingsComposableState {
 /**
  * Methods for working with user settings
  */
-export function useUserSettings(): UseUserSettingsComposableState {
+export default function useUserSettings(): UseUserSettingsComposableState {
```
Comment (slaveeks) [inline src/application/services/useUserSettings.ts:21]:
```suggestion
export function useUserSettings(): UseUserSettingsComposableState {
```

File: src/presentation/components/marketplace/EditorToolElement.vue [hunk 12]
```diff
@@ -20,7 +20,7 @@
 
 <script setup lang="ts">
 import { EditorToolWithUserBinding } from '@/domain/entities/EditorTool';
-import { useUserSettings } from '@/application/services/useUserSettings';
+import useUserSettings from '@/application/services/useUserSettings';
```
Comment (slaveeks) [inline src/presentation/components/marketplace/EditorToolElement.vue:23]:
```suggestion
import { useUserSettings } from '@/application/services/useUserSettings';
```

File: src/presentation/pages/Settings.vue [hunk 13]
```diff
@@ -5,9 +5,14 @@
     v-for="tool in userEditorTools"
     :key="tool.id"
   >
-    <li>
+    <div class="user__tool">
```
Comment (TatianaFomina) [inline src/presentation/pages/Settings.vue:8]: use CSS modules like [here](https://github.com/codex-team/notes.web/blob/main/src/presentation/pages/Home.vue#L2)

Comment (e11sy) [reply src/presentation/pages/Settings.vue]: done

File: src/application/i18n/messages/en.json [hunk 14]
```diff
@@ -10,10 +10,14 @@
       "noteSettings": "Settings"
     }
   },
+  "confirmations": {
+    "uninstallEditorTool": "Do you really want to delete this tool?"
+  },
```
Comment (neSpecc) [inline src/application/i18n/messages/en.json:15]: Let's move it to the `userSettings.toolUninstallConfirmation`

File: src/presentation/pages/Settings.vue [hunk 15]
```diff
@@ -54,12 +61,29 @@ async function userLogout() {
...
 
-<style scoped lang="postcss">
+<style scoped lang="postcss" module>
 @import '@/presentation/styles/typography.pcss';
 
 .marketplace {
   margin-top: var(--spacing-l);
 }
+
+.user__tool {
```
Comment (neSpecc) [inline src/presentation/pages/Settings.vue:84]: if you're using CSS Modules, there is no reason to use BEM. You can use simple classes like `.tool`. CSS Modules will prevent conflict adding hashes to the class under the hood

**Review by elizachi — APPROVED** [review]

File: src/application/i18n/messages/en.json (hunk 14 above)
Comment (e11sy) [reply src/application/i18n/messages/en.json]: done

File: src/presentation/pages/Settings.vue (hunk 15 above)
Comment (e11sy) [reply src/presentation/pages/Settings.vue]: done

**Review by neSpecc — APPROVED** [review]

**Review by TatianaFomina — APPROVED** [review]

**Pull request merged**